source/Cosmos.System2/Network/IPV4/UDP/DHCP/DHCPClient.cs

```diff
@@ -0,0 +1,228 @@
+                DHCPRelease dhcp_release = new DHCPRelease(source, DHCPServerAddress(networkDevice), networkDevice.MACAddress);
+
+                OutgoingBuffer.AddPacket(dhcp_release);
+                NetworkStack.Update();
+
+                NetworkStack.RemoveAllConfigIP();
+
+                IPConfig.Enable(networkDevice, new Address(0, 0, 0, 0), new Address(0, 0, 0, 0), new Address(0, 0, 0, 0));
+            }
+            Close();
+        }
+
+        /// <summary>
+        /// Send a packet to find the DHCP server and tell that we want a new IP address
+        /// </summary>
+        public void SendDiscoverPacket()
+        {
+            NetworkStack.RemoveAllConfigIP();
+
+            foreach (NetworkDevice networkDevice in NetworkDevice.Devices)
+            {
+                IPConfig.Enable(networkDevice, new Address(0, 0, 0, 0), new Address(0, 0, 0, 0), new Address(0, 0, 0, 0));
+
+                DHCPDiscover dhcp_discover = new DHCPDiscover(networkDevice.MACAddress);
+                OutgoingBuffer.AddPacket(dhcp_discover);
+                NetworkStack.Update();
+
+                asked = true;
+            }
+
+            Receive();
+        }
+
+        /// <summary>
+        /// Send a request to apply the new IP configuration
+        /// </summary>
+        private void SendRequestPacket(Address RequestedAddress, Address DHCPServerAddress)
+        {
+            foreach (NetworkDevice networkDevice in NetworkDevice.Devices)
+            {
+                DHCPRequest dhcp_request = new DHCPRequest(networkDevice.MACAddress, RequestedAddress, DHCPServerAddress);
+                OutgoingBuffer.AddPacket(dhcp_request);
+                NetworkStack.Update();
+            }
+            Receive();
+        }
+
+        /*
+         * Method called to applied the differents options received in the DHCP packet ACK
+         **/
+        /// <summary>
+        /// Apply the new IP configuration received.
+        /// </summary>
+        /// <param name="Options">DHCPOption class using the packetData from the received dhcp packet.</param>
+        /// <param name="message">Enable/Disable the displaying of messages about DHCP applying and conf. Disabled by default.
+        /// </param>
+        private void Apply(DHCPAck packet, bool message = false)
+        {
+            NetworkStack.RemoveAllConfigIP();
+
+            //cf. Roadmap. (have to change this, because some network interfaces are not configured in dhcp mode) [have to be done in 0.5.x]
+            foreach (NetworkDevice networkDevice in NetworkDevice.Devices)
+            {
+                if (packet.Client.ToString() == null ||
+                    packet.Client.ToString() == null ||
+                    packet.Client.ToString() == null ||
+                    packet.Client.ToString() == null)
+                {
+                    Global.mDebugger.Send("Parsing DHCP ACK Packet failed, can't apply network configuration.");
```

Comment:
Could the kernel want to know this? If yes either return something or throw an error.